### PR TITLE
stream functions

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/practice/CandidateValidator.java [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+                && isLongTimePeriod(candidate.getPeriodsInUkr());
+    }
+
+    private boolean isLongTimePeriod(String prePeriod) {
+        String[] periodInUkr = prePeriod.split("-");
+        int countYearInUkr = Integer.parseInt(periodInUkr[1]) - Integer.parseInt(periodInUkr[0]);
+        if (countYearInUkr >= 10) {
+            return true;
+        }
+        return false;
```
Comment (irynamekh) [inline src/main/java/practice/CandidateValidator.java:21]: write like this, you don't need if
return countYearInUkr >= 10 (magic num)

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -23,7 +33,29 @@ public int findMinEvenNumber(List<String> numbers) {
...
+                        num -= 1;
+                    }
+                    return num;
+                })
+                .collect(Collectors.toList());
+
+        List<Integer> oddNumbers = oddIndexNumbers.stream()
+                .filter(num -> num % 2 != 0)
+                .collect(Collectors.toList());
+
```
Comment (irynamekh) [inline src/main/java/practice/StreamPractice.java:49]: you have many redundant empty lines

**Review by irynamekh — CHANGES_REQUESTED** [review]

**Review by irynamekh — APPROVED** [review]

File: src/main/java/practice/StreamPractice.java (hunk 2 above)
Comment (Vasul-Pankiv) [reply src/main/java/practice/StreamPractice.java:52]: not resolved

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -23,7 +36,29 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> oddIndexNumbers = IntStream.range(0, numbers.size())
+                .mapToObj(i -> {
+                    int num = numbers.get(i);
+                    if (i % MOD_NUM_CHECK != ZERO_NUM) {
```
Comment (Vasul-Pankiv) [inline src/main/java/practice/StreamPractice.java:42]:
```suggestion
                    if (i % MOD_NUM_CHECK != ZERO_NUM) {
```
"i % MOD_NUM_CHECK" extract it to method and reuse

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request closed without merging**